src/shared/yagna/adapters/network-api-adapter.ts

```diff
@@ -0,0 +1,37 @@
+import { YagnaApi } from "../yagnaApi";
+import { Logger } from "../../utils";
+import { INetworkApi } from "../../../network/api";
+import { Network, NetworkNode } from "../../../network";
+
+export class NetworkApiAdapter implements INetworkApi {
+  constructor(
+    private readonly yagnaApi: YagnaApi,
+    private readonly logger: Logger,
+  ) {}
+
+  async createNetwork(options: { id: string; ip: string; mask?: string; gateway?: string }): Promise<Network> {
+    const { id, ip, mask, gateway } = await this.yagnaApi.net.createNetwork(options);
+    // @ts-expect-error TODO: Can we create a network without an id or is this just a bug in ya-clinet spec?
+    return new Network(id, ip, mask, gateway);
+  }
+  async removeNetwork(network: Network): Promise<void> {
+    return this.yagnaApi.net.removeNetwork(network.id);
+  }
+  async createNetworkNode(network: Network, nodeId: string, nodeIp: string): Promise<NetworkNode> {
+    await this.yagnaApi.net.addNode(network.id, { id: nodeId, ip: nodeIp });
+    return new NetworkNode(nodeId, nodeIp, network.getNetworkInfo.bind(network));
+  }
+  async removeNetworkNode(network: Network, node: NetworkNode): Promise<void> {
+    await this.yagnaApi.net.removeNode(network.id, node.id);
+  }
+
+  async getIdentity() {
+    return this.yagnaApi.identity.getIdentity().then((res) => res.identity);
+  }
```

Comment:
I think calls to yagna api should be wrapped in try-catch blocks and mapped to our domain errors. At least that's what I did in all other adapters